horiz_interp/include/horiz_interp_bicubic.inc

```diff
@@ -533,25 +452,35 @@ module horiz_interp_bicubic_mod
 !      ansy2=ansy2/(x2u-x2l) ! could be used for accuracy checks
       return
 !  (c) copr. 1986-92 numerical recipes software -3#(-)f.
-   end subroutine bcuint
+   end subroutine BCUINT_
 !---------------------------------------------------------------------------
 
-   subroutine bcucof(y,y1,y2,y12,d1,d2,c)
-      real d1,d2,c(4,4),y(4),y1(4),y12(4),y2(4)
+   subroutine BCUCOF_(y,y1,y2,y12,d1,d2,c)
+      real(FMS_HI_KIND_) d1,d2,c(4,4),y(4),y1(4),y12(4),y2(4)
       integer i,j,k,l
-      real d1d2,xx,cl(16),wt(16,16),x(16)
-      save wt
-      data wt/1., 0., -3., 2., 4*0., -3., 0., 9., -6., 2., 0., -6., 4., 8*0., &
-              3., 0., -9., 6., -2., 0., 6., -4., 10*0., 9., -6., 2*0., -6., &
-              4., 2*0., 3., -2., 6*0., -9., 6., 2*0., 6., -4., 4*0., 1., 0., &
-              -3., 2., -2., 0., 6., -4., 1., 0., -3., 2., 8*0., -1., 0., 3., &
-              -2., 1., 0., -3., 2., 10*0., -3., 2., 2*0., 3., -2., 6*0., 3., &
-              -2., 2*0., -6., 4., 2*0., 3., -2., 0., 1., -2., 1., 5*0., -3., &
-              6., -3., 0., 2., -4., 2., 9*0., 3., -6., 3., 0., -2., 4., -2., &
-              10*0., -3., 3., 2*0., 2., -2., 2*0., -1., 1., 6*0., 3., -3., &
-              2*0., -2., 2., 5*0., 1., -2., 1., 0., -2., 4., -2., 0., 1., -2., &
-              1., 9*0., -1., 2., -1., 0., 1., -2., 1., 10*0., 1., -1., 2*0., &
-              -1., 1., 6*0., -1., 1., 2*0., 2., -2., 2*0., -1., 1./
+      real(FMS_HI_KIND_) d1d2,xx,cl(16),x(16)
+      integer, parameter :: kindl = FMS_HI_KIND_!< compiled kind type
+      !! previous set by a data block, zeroing out since last few rows werent st
+      real(FMS_HI_KIND_), save, dimension(16,16) :: wt
+      wt = 0.0_kindl
+      wt(1,:) = (/1.0_kindl, 0.0_kindl, -3.0_kindl, 2.0_kindl, 4*0.0_kindl, -3.0_kindl, 0.0_kindl, 9.0_kindl, &
```

Comment:
Yeah we looked at this in our meeting and weren't really sure why they were doing that either but you're actually right with the 4 0.0's in a row lol. I found a c version of this algorithm online and the weights only matched up with that. Apparently that actually works in a data construct, `0.0*4` will put 4 consecutive 0.0's in the array instead of multiplying... i hate fortran who thought this syntax was a good idea